sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSQLInsertTestSuite.scala

```diff
@@ -21,5 +21,20 @@ import org.apache.spark.sql.SQLInsertTestSuite
 import org.apache.spark.sql.hive.test.TestHiveSingleton
 
 class HiveSQLInsertTestSuite extends SQLInsertTestSuite with TestHiveSingleton {
+
+  private val originalPartitionMode = spark.conf.getOption("hive.exec.dynamic.partition.mode")
```

Comment:
in sql/core, the `SparkSession` is created for each test suite, and we can't use `spark` in the class constructor. In sql/hive it's OK as all test suites share one global `SparkSession`, but let's avoid using this pattern that only works in sql/hive.

Reply:
Oh right, this should be avoided